runner/utils/driver/docker/helpers.go

```diff
@@ -20,45 +20,62 @@ func getServiceDomain(projectID, serviceID string) string {
 }
 
 func (d *Docker) pullImageIfDoesntExists(ctx context.Context, projectID string, taskDocker model.Docker) error {
-	_, _, err := d.client.ImageInspectWithRaw(ctx, taskDocker.Image)
-	if err != nil {
-		// image doesn't exist locally
-		if taskDocker.Secret != "" {
-			data, err := ioutil.ReadFile(fmt.Sprintf("%s/%s/%s.json", d.secretPath, projectID, taskDocker.Secret))
+	if taskDocker.ImagePullPolicy == model.PullIfNotExists {
+		_, _, err := d.client.ImageInspectWithRaw(ctx, taskDocker.Image)
+		if err != nil {
```

Comment:
Simply return if error is nil. You are pulling the image down there anyways

Reply:
You didn't do this @KavishShah09

Reply:
At first I thought you were talking about the error messages.
Got it now what you meant